src/main/java/com/example/dr_app/service/PatientService.java

```diff
@@ -30,4 +31,21 @@ public Patient getPatientData(Long id) {
         Optional<Patient> patientOptional = patientRepository.findById(id);
         return patientOptional.get();
     }
+
+    public List<Patient> getPatientsByGenderData(PatientGender gender) {
+        return patientRepository.findByGender(gender);
+    }
+    public List<Patient> getMalePatientsData() {
```

Comment:
I believe this can be refactored, you dont need three methods, this can be achieved by one method. getPatientsByGenderData and as the argument you can pass the Enum.